### ISSUE TITLE
darknet detecting far to many objects and in random locations

### ISSUE THREAD

**Issue by maribethrc** [pjreddie/darknet#525]
Background: I'm training yolov2 on 3 new classes (about 150 images per class), following the VOC guide on the yolo website. I've read as many guides/issues as I can find and followed all the instructions. I've changed the following: 

.cfg file, classes = 3 and filters = 40 (last layer)
.data file, paths are to my data locations
yolo.c, I changed the *voc_names[] to the names of my 3 classes, and changed the number of classes to 3 in the call to draw_detections().

When I train the network on my 3 classes, I do not get any errors.. but the loss hovers around 300 without moving and Obj/No Obj always stay around 0.5 (i.e., it's not learning). I trained to 3000 steps on the full dataset, and I also trained on a small set of 3 images/class to intentionally see if I could overfit. Also, when trying to overfit, I made my testing and training images the same (it should be really easy to detect what you trained on..) Both times, no learning seems to be happening.

When I test with the weights generated from training with all the data, I get no bounding boxes.

When I test the network with the weights from training on only 3 images/class, I get TONS of object predictions, none of which make sense... often they are just bounding white space.

So my question... what am I missing?? Has anyone else had these problems? I've scrounged the web for what else I might need to change without luck.. 

The image below from testing the network (using weights from training on only 3 images/class) is after setting -thresh 0.9
![predictions](https://user-images.githubusercontent.com/37320540/37322688-1bea971e-264d-11e8-9c7b-b41d23dde9bc.jpg)

**Comment by ahsan856jalal** [comment]
I suggest you to use AlexeyAB's fork of darknet. It is more stable and has better support as I just shifted to that due to similar sort of problems.
[https://github.com/AlexeyAB/darknet](url)

**Comment by ahsan856jalal** [comment]
But make sure that you have made text files in a correct way
label x y w h
label=0-(N-1)
x=midpoint_x/img_width
y=midpoint_y/img_height
w=box_width/img_width
h=box_height/img_height

**Comment by TheMikeyR** [comment]
Also it is recommended to have at least 2000 images per class (just a recommendation, it could still be possible to train with less)

**Comment by maribethrc** [comment]
Yes, my text files were correct. 
I switched to AlexeyAB's fork and that fixed my problems, it is training and detecting fine now. Wish I'd tried that earlier! Thanks for the help.
